src/Parser/ParserInterface.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace React\Http\Parser;
+
+use Evenement\EventEmitterInterface;
+use React\Http\Request;
+
+interface ParserInterface extends EventEmitterInterface
+{
+    public function __construct(Request $request);
```

Comment:
The constructor must not be part of the interface. What is the intend here? Perhaps renaming this to a method also makes this more obvious?